crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -33,6 +33,9 @@ class ApplicationController < ActionController::Base
     Crowbar::Installer.successful? || \
     Rails.env.test?
   }
+  before_filter :upgrade, if: proc {
+    File.exist?("/var/lib/crowbar/upgrade/6-to-7-progress.yml")
```

Comment:
That's not enough. We are not deleting `6-to-7-progress.yml` after the upgrade is done, we're touching `/var/lib/crowbar/upgrade/6-to-7-upgraded-ok` once the upgrade is finished...